File: 2801-2900/t2894.cpp
// 2025-05-27

class Solution {
public:
    int differenceOfSums(int n, int m) {
        int res = 0;
        for (int i = 1; i <= n; ++i) {
            if (i % m) res += i;
            else res -= i;
        }
        return res;
    }
};

File: 2801-2900/t2894.py
# 2025-05-27

class Solution:
    def differenceOfSums(self, n: int, m: int) -> int:
        cnt = n // m
        num2 = (m + m * cnt) * cnt / 2
        num1 = (1 + n) * n / 2 - num2
        return int(num1 - num2)
